### PR TITLE
Mention version scheme in readme

### PR DESCRIPTION
**Pull request by tulinkry**
I added a note about opengrok version meaning when doing an update

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,6 +30,14 @@ Official page of the project is on:
 
 See https://github.com/oracle/opengrok/wiki/How-to-setup-OpenGrok
 
+### 2. 1. Updating
+
+OpenGrok uses [semantic versioning](https://semver.org/) and the version components further tells more details about updating to newer version. The version scheme is x.y.z and change in any component is interpreted as:
```
Comment (idodeclare) [inline README.md:35]: Maybe better as "tell"?

**Review by idodeclare — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (tulinkry) [reply README.md:35]: indicates

File: README.md [hunk 2]
```diff
@@ -30,6 +30,14 @@ Official page of the project is on:
 
 See https://github.com/oracle/opengrok/wiki/How-to-setup-OpenGrok
 
+### 2. 1. Updating
+
+OpenGrok uses [semantic versioning](https://semver.org/) and the version components further indicate more details about updating to newer version. The version scheme is x.y.z and change in any component is interpreted as:
+
+ - x - major backwards incompatible update
+ - y - full clean reindex of your repositories is needed (e. g. index format has changed)
+ - z - redeployment of the web application is necessary
```
Comment (vladak) [inline README.md:39]: not in all the cases

Comment (tulinkry) [reply README.md:39]: the wiki says only:

> If only z changes, just redeploy new web application and keep going.

Comment (vladak) [reply README.md:39]: That's recommended, not necessary.

Comment (vladak) [inline README.md:39]:
```suggestion
 - z - redeploy web application
```

Comment (tulinkry) [reply README.md]: ah ok

**Pull request merged**